src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/IKestrelTrace.cs

```diff
@@ -39,5 +39,7 @@ public interface IKestrelTrace : ILogger
         void NotAllConnectionsClosedGracefully();
 
         void ApplicationError(Exception ex);
```

Comment:
LoggingThreadPool. @benaadams worked hard to making logging lower allocation, so I wanted his feedback on how to do it. It looks like one LoggingThreadPool per connection might be the solution.

LoggingThreadPool is also used by KestrelThread so it might not be possible to log connection Id's there, but the queued callbacks should never throw all the way up to KestrelThread, so it's really just for safety. For those maybe we should add an UnexpectedKestrelError log since it's likely not an application error.